sql/driver/conn.go

```diff
@@ -111,6 +108,12 @@ func (c *conn) send(args Request) (*rows, error) {
 	if index < 0 {
 		return r, nil
 	}
+	// Check for any application errors.
+	for _, result := range resp.Results {
+		if result.Error != nil {
+			return nil, result.Error
```

Comment:
Another option to returning the first error is to make a custom error type which holds all of the individual errors and the index of the result they came from. Probably not worth doing it at this point, but may be worth a comment.